src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -97,12 +102,19 @@ class StreamReader {
 
 private:
     // Constructor
-    explicit StreamReader(StreamReaderDataContext<clp::ir::four_byte_encoded_variable_t>&&
-                                  stream_reader_data_context);
+    explicit StreamReader(
+            StreamReaderDataContext<four_byte_encoded_variable_t>&& stream_reader_data_context
+    );
+
+    // Methods
+    [[nodiscard]] static auto create_deserializer_and_data_context(
```

Comment:
How about `create_data_context`? Since the deserializer is created and directly moved into the context, I don't think we need to mention it in the name.